src/context.c

```diff
@@ -52,6 +53,20 @@ static const gchar* get_cmdline_bootname(void)
 			return bootname;
 	}
 
+	/* Support NVIDIA's CBoot bootloader.
+	*
+	*  If /proc/cmdline contains an entry for boot.slot_suffix= with an empty value,
+	*  it implies that A partition is selected. If the value is '_b' then the system
+	*  was booted from B parition. */
+	suffix = regex_match("boot\\.slot_suffix=((\\s+|\\S+))", contents);
```

Comment:
Just a guess: If we stick to the idea of introducing a configurable key (instead of fixed `boot.slot_suffic`) here and allow it to match an empty parameter by default, wouldn't that work, too? Maybe we forward that internally as a predefined value, like '`empty`' to allow distinguishing it from an actually empty boot name variable in the further processing (which might be an error in other cases).

@jluebbe What's your opinion on that?

Reply:
Are you suggesting to turn the `bootname` key in `system.conf` into a regex? This would make the matching scheme more flexible I guess.

Reply:
Another option would be to use a similar scheme as in `bootchooser.c`: execute a shell script when bootloader is set to `custom`.